source/elements/oneDAL/source/data_management/tables.rst

```diff
@@ -1,93 +1,88 @@
 .. highlight:: cpp
 
+.. _tables:
+
 ======
 Tables
 ======
 
-Table is a generic |dal_short_name| concept over numerical data. It provides uniformed way
-to pass the data to the library as :ref:`inputs <Input>` or :ref:`parameters <Descriptor>` or get them as :ref:`results <Result>`.
+This section describes the types related to the :ref:`table <table>` concept.
+|dal_short_name| defines types of the table and the :ref:`metadata <metadata>`
+concepts associated with it:
 
-Table object is a container of two entities: data and metadata.
+- The :code:`table` is a base class that implements the table concept and
+  provides its metadata. Each implementation of the :ref:`table <table>` concept
+  shall be derived from the :code:`table` class (for more details, see
+  :ref:`table_api`).
 
-- Data is organized in a shape of :math:`(N \times p)`,
-  where :math:`N` is a number of :term:`observations <Observation>` in a table and :math:`p`
-  is a number of :term:`features <Feature>`.
+- The :code:`table_meta` class implements the metadata concept for the
+  table. This type can be extended in the types derived from the table (for more
+  details, see :ref:`Metadata API`).
 
-- Metadata defines the detailed structure of the data and,
-  therefore, helps |dal_short_name| to access the data efficiently
-  (see :ref:`Metadata API` section for details).
 
--------------------------------
-Table types in |dal_short_name|
--------------------------------
-|dal_short_name| defines a set of classes, each class implements a table contract
-with a specific set of metadata values (see :ref:`Metadata API` for details):
+------------
+Requirements
+------------
 
-.. list-table::
-  :header-rows: 1
-
-  * - Table type
-    - :ref:`Data layout <Data layout API>`
-    - :ref:`Data format <Data format API>`
-    - Is contiguous
-    - Is homogeneous
-  * - homogen_table_
-    - row_major/column_major
-    - dense
-    - yes
-    - yes
-  * - soa_table_
-    - column_major
-    - dense
-    - yes
-    - yes/no
-  * - aos_table_
-    - row_major
-    - dense
-    - yes
-    - yes/no
-  * - csr_table_
-    - row_major
-    - csr
-    - yes
-    - yes
-
------------------------------
-Requirements on table objects
------------------------------
-Each table object in |dal_short_name| follows these requirements:
-
-1. Table objects in |dal_short_name| are :term:`immutable <Immutability>` (it is not possible
-   to change data or metadata values inside the table).
-
-2. To create complex table types or modify table data, :ref:`builders <Builders>` should be used.
-
-3. Table objects in |dal_short_name| are reference-counted. One can use an assignment operator or copy constructor
-   on table objects to create another reference to it.
-   ::
+Each type that implements :ref:`table <table>` concept in |dal_short_name|
```

Comment:
each table type?

Reply:
`each table table type that implements table concept` sounds odd, too many "tables". Need to cleanup the mess with concepts/types/implementations after 0.8